src/library/scala/concurrent/Future.scala

```diff
@@ -566,18 +567,29 @@ object Future {
    */
   object never extends Future[Nothing] {
 
-    private[this] final val notGoingToHappen = new CountDownLatch(1)
-
     @throws[TimeoutException]
     @throws[InterruptedException]
     override final def ready(atMost: Duration)(implicit permit: CanAwait): this.type = {
       import Duration.{Undefined, Inf, MinusInf}
       atMost match {
         case u if u eq Undefined => throw new IllegalArgumentException("cannot wait for Undefined period")
-        case `Inf`               => notGoingToHappen.await()
+        case `Inf`               =>
+          while(!Thread.interrupted()) {
+            LockSupport.park(this)
```

Comment:
A side effect of this will be that "monitoring and diagnostic tools" that observe `LockSuppoer.getBlocker` will see `object never` rather than the defunct `notGoingToHappen` `CountdownLatch`. I don't know what sort of tools use this, do you, @viktorklang? I doubt the change  will hurt, I'm just curious :)

Reply:
AFAIR variable name doesn't go into the thread dump or `LockSupport.getBlocker`. What is available and shown is the class name and instance address. Therefore I don't think `notGoingToHappen` is showing now anywhere in thread dumps.

Reply:
```
scala> import scala.concurrent._
import scala.concurrent._

scala> Await.result(Future.never, duration.Duration.Inf)
```

Status Quo:
```
"main" #1 prio=5 os_prio=31 tid=0x00007fef2d004000 nid=0x1d03 waiting on condition [0x0000700000fc4000]
   java.lang.Thread.State: WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x00000007bc030260> (a java.util.concurrent.CountDownLatch$Sync)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:836)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(AbstractQueuedSynchronizer.java:997)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1304)
	at java.util.concurrent.CountDownLatch.await(CountDownLatch.java:231)
	at scala.concurrent.Future$never$.ready(Future.scala:570)
	at scala.concurrent.Future$never$.result(Future.scala:580)
	at scala.concurrent.Future$never$.result(Future.scala:563)
	at scala.concurrent.Await$.$anonfun$result$1(package.scala:215)
	at scala.concurrent.Await$$$Lambda$1188/187540389.apply(Unknown Source)
	at scala.concurrent.BlockContext$DefaultBlockContext$.blockOn(BlockContext.scala:53)
	at scala.concurrent.Await$.result(package.scala:142)
	at $line14.$read$$iw$$iw$$iw$$iw$.<init>(<console>:15)
```

This PR:

```
"main" #1 prio=5 os_prio=31 tid=0x00007fe94a800800 nid=0x2303 waiting on condition [0x0000700004310000]
   java.lang.Thread.State: WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x000000076c308398> (a scala.concurrent.Future$never$)
```

Seems like an improvement to me!

Reply:
@retronym Thanks, Jason! Definitely beyond just fixing the Scala.js regression ^^